src/rate_limiter/src/lib.rs

```diff
@@ -224,6 +231,58 @@ impl TokenBucket {
     }
 }
 
+/// State for saving a TokenBucket.
+#[derive(Versionize)]
+pub struct TokenBucketState {
```

Comment:
nit: I would add a separate file/module for serialization in general.

Reply:
will do.